src/System.Net.Security/src/System/Net/Security/SecureChannel.cs

```diff
@@ -407,7 +403,7 @@ private bool AcquireClientCredentials(ref byte[] thumbPrint)
                 }
                 else
                 {
-                    if (ClientCertificates.Count == 0)
+                    if (_clientCertificates == null || _clientCertificates.Count == 0)
```

Comment:
Would it be a good idea to flip the if and change to if(_clientCertificates?.Count > 0) ?